### PR TITLE
Windows: Fixing longpath hanlding of UNC paths.

### PR DESCRIPTION
**Pull request by swernli**
Signed-off-by: Stefan J. Wernli swernli@microsoft.com

### CODE REVIEW

**Comment by swernli** [comment]
@jhowardmsft @jstarks @tiborvass @jfrazelle @stevvooe
Forgot to tag folks in the PR description.  This fixes the longpath handling to correctly account for UNC paths.  Current changes break docker build from UNC share, and this update fixes that regression.

**Comment by jstarks** [comment]
LGTM

**Comment by tiborvass** [comment]
@swernli needs unit test

File: pkg/longpath/longpath.go [hunk 1]
```diff
@@ -15,7 +15,12 @@ const Prefix = `\\?\`
 // it does not already have it.
 func AddPrefix(path string) string {
 	if !strings.HasPrefix(path, Prefix) {
-		path = Prefix + path
+		if strings.HasPrefix(path, `\\`) {
+			// This is a UNC path, so we need to add 'UNC' to the path as well.
+			path = Prefix + `UNC` + path[1:]
```
Comment (tiborvass) [inline pkg/longpath/longpath.go]: So `\\foo` is supposed to become `\\?\UNC\foo`, correct?

Comment (swernli) [inline pkg/longpath/longpath.go]: Yes, that's correct.

**Comment by swernli** [comment]
Working on a simple unit test...

**Comment by swernli** [comment]
@jhowardmsft Any ideas why this is failing?  I can't make heads or tails of the janky output...

**Comment by jhowardmsft** [comment]
@swernli - wow! No, you've hit one I've never seen before. @jfrazelle Any ideas? 

`Precompiled: github.com/docker/docker/daemon
# cover github.com/docker/docker/daemon
cover: internal error: block 197 overlaps block 198
	/go/src/github.com/docker/docker/daemon/container.go:#22630,#22722 /go/src/github.com/docker/docker/daemon/container.go:#22668,#22707`

**Comment by swernli** [comment]
Finally got this into a good state.  Unit test is passing, too.  Any other feedback, or is this ready to merge?

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**